### PR TITLE
Add boot rank to OSImage interface.

### PR DESCRIPTION
**Pull request by terry2042002**
The rank represents the boot order of the images. The higher the
rank, the prior the image shows in the boot menu.

We set the BLS block to a higher default rank and add a command line flag
"BLS_BOOT_RANK" to control the boot order.

Signed-off-by: Terry Tang <terrytang@google.com>

### CODE REVIEW

File: pkg/boot/boot.go [hunk 1]
```diff
@@ -21,6 +21,9 @@ type OSImage interface {
 	// Label is intended for boot menus.
 	Label() string
 
+	// Rank the priority of the images for boot menus.
```
Comment (rjoleary) [inline pkg/boot/boot.go:24]: Add a comment here saying a larger integer takes priority.

**Review by rjoleary — COMMENTED** [review]
Squash commits and add explanation on why BLS needs a higher rank by default.

File: pkg/boot/bls/bls.go [hunk 2]
```diff
@@ -19,14 +19,16 @@ import (
...
 	"strings"
 
 	"github.com/u-root/u-root/pkg/boot"
 	"github.com/u-root/u-root/pkg/ulog"
 )
 
 const (
-	blsEntriesDir = "loader/entries"
+	blsEntriesDir  = "loader/entries"
+	blsDefaultRank = 1
```
[the remark this answers is not in the document]
Comment (terry2042002) [reply pkg/boot/bls/bls.go:33]: Done

File: pkg/boot/localboot/localboot.go [hunk 3]
```diff
@@ -18,6 +19,12 @@ import (
 	"github.com/u-root/u-root/pkg/ulog"
 )
 
+type byRank []boot.OSImage
```
[the remark this answers is not in the document]
Comment (terry2042002) [reply pkg/boot/localboot/localboot.go:23]: Done

File: pkg/boot/boot.go (hunk 1 above)
Comment (terry2042002) [reply pkg/boot/boot.go:24]: Done

**Review by rminnich — APPROVED** [review]

**Review by rminnich — APPROVED** [review]

**Review by rjoleary — APPROVED** [review]

**Pull request merged**